backend/src/main/java/eu/clarin/switchboard/resources/DataResource.java

```diff
@@ -129,10 +126,59 @@ public Response getFile(String idString, String mediatype) throws Throwable {
                 mediatype = mediatype + ";charset=utf-8";
             }
         }
-        return Response.ok(fileStream)
-                .type(mediatype)
-                .header("content-disposition", "attachment; filename=" + fi.getFilename())
-                .build();
+
+        if (range == null || range.isEmpty()) {
+            StreamingOutput fileStream = output -> {
+                byte[] data = Files.readAllBytes(fi.getPath());
+                output.write(data);
+                output.flush();
+            };
+            return Response.ok(fileStream)
+                    .type(mediatype)
+                    .header("content-disposition", "attachment; filename=" + fi.getFilename())
+                    .build();
+        }
+
+        String[] ranges = range.split("=");
```

Comment:
Might be an idea to do a regex based validation before the processing of the range specification? As an extra, quick check to know for sure that we have `bytes=[0-9]+-[0-9]+` and nothing else? Would prevent risk of failed integer parse.